rclpy/test/test_client.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2017 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+import unittest
+
+from rcl_interfaces.srv import GetParameters
```

Comment:
rclcpp defines [mocking interfaces](https://github.com/ros2/rclcpp/tree/master/rclcpp/test/mock_msgs) to test this kind of things rather than relying on existing rcl_interfaces. It may be worth doing the same things here to avoid confusion as an outsider would expect Parameters to be a thing that exists in rclpy when seeing this service being used

Reply:
Mind if this goes in a follow up PR? `test_callback_group.py`, `test_destruction.py`,  and `test_node.py` would also benefit from this.